src/webots/nodes/WbTransform.cpp

```diff
@@ -52,6 +60,24 @@ WbTransform::~WbTransform() {
     wr_node_delete(WR_NODE(wrenNode()));
 }
 
+void WbTransform::reset(const QString &id) {
+  WbGroup::reset(id);
+  // note: for solids, the setting of these parameters has to occur only if mJointParents.size() == 0 and is taken care of in
+  // WbSolid::reset, otherwise it breaks the resetting of hinge based joints
```

Comment:
```suggestion
  // note: for solids, the set of these parameters has to occur only if mJointParents.size() == 0 and it is handled in
  // WbSolid::reset, otherwise it breaks the reset of hinge based joints
```